swim.go

```diff
@@ -63,28 +79,82 @@ func (s *SWIM) ShutDown() {
 //    The member(j) is judged to be failed, so check the member(j) as suspected or delete the member(j) from memberMap.
 //
 // ** When performing ping, ack, and indirect-ping in the above procedure, piggybackdata is sent together. **
+//
+//
+// startFailureDetector function
+//
+// 1. Pick a member from memberMap.
+// 2. Probe the member.
+// 3. After finishing probing all members, reset memberMap
+//
 func (s *SWIM) startFailureDetector() {
 
-}
+	go func() {
+		for {
+			// Get copy of current members from memberMap.
+			members := s.memberMap.GetMembers()
+			for _, member := range members {
+				s.probe(member)
+			}
 
-// Delegate interface to notify status change of the member
-type MemberStatusChangeDelegate interface {
-	deleteMember(id MemberID)
-	addMember()
-	updateMember()
+			// Reset memberMap.
+			s.memberMap.reset()
+		}
+	}()
```

Comment:
I think there's no T time interval. You may skipped because this is sketch. But I think it's better to comment there will be T time interval